### ISSUE TITLE
ITransformProvider2 is missing

### ISSUE THREAD

**Issue by dotMorten** [dotnet/wpf#10847]
Windows 10 added a better pattern for ITransformProvider, ITransformProvider2.

That pattern is missing from WPF, limiting the accessibility features of WPF compared to UWP/WinUI.

Here's the equivalent UWP/WinUI3 version:
https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.provider.itransformprovider2?view=winrt-26100

**Comment by miloush** [comment]
There has been plenty of automation interfaces added, do you have a specific requirement for this one?

**Comment by lindexi** [comment]
@dotMorten Is it only the [MapControl](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.peers.mapcontrolautomationpeer?view=winrt-26100) uses ITransformProvider2 ?

**Comment by dotMorten** [comment]
@lindexi  Any custom control that allows zoom and pan will need this interface. In my case yes I'm building a Map Control and want to make it accessible. In my WinUI version I can now use full voice control to navigate the map just by implementing this pattern - I'm not able to provide the same experience in the WPF version.

**Comment by dotMorten** [comment]
@lindexi I've started a PR with most of the work done - however there new UIAutomation APIs doesn't support the old (and now deprecated) way of calling into `uiautomationcoreapi.h`, so will need a slightly different approach, that I'm not currently quite sure how to do yet. See the notes in the PR.

**Comment by lindexi** [comment]
@dotMorten I read your code and I think your definition is in line with the documentation, but I don't know how to do it too. In addition, I am curious about how behaviors such as Zoom should properly drive the interaction of the UI interface. Since WPF does not have a built-in MapControl control, I don't know which existing controls I can refer to

**Comment by dotMorten** [comment]
> Since WPF does not have a built-in MapControl control, I don't know which existing controls I can refer to

I'm not following this argument. There are no existing controls to refer to because the pattern isn't implemented, so no controls would be using it.
Also the notion that this is only for maps is a bit odd. There are lots of other things that supports zooming, like a text viewer, an image editor etc.

I just built a custom control that exposes an automation peer which implements the pattern. This is not quite working yet, since the interop for the new pattern is different than the older (deprecated) C-APIs that they use (as pointed out by the TODO in my PR and why this is still in-progress).

**Comment by lindexi** [comment]
@dotMorten You are right. Such as the image editor and so on.
